lib/event-stream.js

```diff
@@ -269,4 +296,34 @@ EventStream.prototype._read = function() {
 	this.getLongPollInfo();
 };
 
+/**
+ * Implementation of stream-internal `_destroy` function (v8.0.0 and later).
+ * Called by stream consumers to effectively stop polling via the public
+ * `destroy()`.
+ * @returns {void}
+ * @private
+ */
+EventStream.prototype._destroy = function() {
+	clearTimeout(this._retryTimer);
+	delete this._retryTimer;
+};
+
+// backwards-compat for Node.js pre-v8.0.0
+if (!(typeof Readable.destroy === 'function')) {
```

Comment:
You'll need to check against `Readable.prototype.destroy` here — this condition as written will always include the polyfill.  Also, using `!==` might be more readable than `!(... === ...)`

Reply:
my bad